txscript/standard.go

```diff
@@ -485,28 +520,31 @@ func isNullDataScript(scriptVersion uint16, script []byte) bool {
 // NOTE:  All scripts that are not version 0 are currently considered non
 // standard.
 func typeOfScript(scriptVersion uint16, script []byte) ScriptClass {
-	if scriptVersion != 0 {
-		return NonStandardTy
-	}
-
-	switch {
-	case isPubKeyScript(script):
-		return PubKeyTy
-	case isPubKeyHashScript(script):
-		return PubKeyHashTy
-	case isScriptHashScript(script):
-		return ScriptHashTy
-	case isWitnessPubKeyHashScript(script):
-		return WitnessV0PubKeyHashTy
-	case isWitnessScriptHashScript(script):
-		return WitnessV0ScriptHashTy
-	case isMultisigScript(scriptVersion, script):
-		return MultiSigTy
-	case isNullDataScript(scriptVersion, script):
-		return NullDataTy
-	default:
-		return NonStandardTy
+	switch scriptVersion {
+	case 0:
+		switch {
+		case isPubKeyScript(script):
+			return PubKeyTy
+		case isPubKeyHashScript(script):
+			return PubKeyHashTy
+		case isScriptHashScript(script):
+			return ScriptHashTy
+		case isWitnessPubKeyHashScript(script):
+			return WitnessV0PubKeyHashTy
+		case isWitnessScriptHashScript(script):
+			return WitnessV0ScriptHashTy
+		case isMultisigScript(scriptVersion, script):
+			return MultiSigTy
+		case isNullDataScript(scriptVersion, script):
+			return NullDataTy
+		}
+	case 1:
+		switch {
+		case isWitnessTaprootScript(script):
+			return WitnessV1TaprootTy
+		}
 	}
```

Comment:
I guess this isn't really exercised since `typeOfScript` is only used in `CalcScriptInfo`, which is now clear defined only for v0 scripts. Should I leave?